### PR TITLE
[libcxx] Add OHOS support

### PR DESCRIPTION
**Pull request by kpdev**
Add support for OpenHarmony OS (OHOS)

OpenHarmony is an open-source project incubated and operated by the OpenAtom Foundation. It targets IoT, and already used on production devices.

General OpenHarmony OS discussion on discourse thread:
  "Add support for OpenHarmony OS"
  https://discourse.llvm.org/t/rfc-add-support-for-openharmony-os/66656

LLVM and clang parts of OHOS support: D138202, D145227

Phabricator review: D146804

~~

Huawei RRI, OS Lab

### CODE REVIEW

**Review by mordante — COMMENTED** [review]
In general I'm happy with the changes.
Our build system has been changed quite a bit recently. Is your build executed?

File: libcxx/utils/ci/BOT_OWNERS.txt [hunk 1]
```diff
@@ -20,3 +20,8 @@ N: Android libc++
 E: rprichard@google.com
 H: rprichard
 D: Emulator-based x86[-64] libc++ CI testing
+
+N: OpenHarmony libc++
+E: kosov.pavel@huawei.com
+H: kpdev42
+D: AArch64
```
[the remark this answers is not in the document]
Comment (kpdev) [reply libcxx/utils/ci/BOT_OWNERS.txt]: Done

**Comment by kpdev** [comment]
> In general I'm happy with the changes. Our build system has been changed quite a bit recently. Is your build executed?

Thank you for the review ) Actually, I've checked the build agent status (on buildkite.com), and it is connected, but "This agent has not run any jobs". Looks like I setup something wrong (it is my first time of buildkite setup, I am sorry). Maybe there are some guides/examples of build agent setup? I checked this link https://libcxx.llvm.org//AddingNewCIJobs.html#addingnewcijobs and presentation from Louis https://www.youtube.com/watch?v=B7gB6van7Bw , but whole picture of this setup is still missing for me )

Now I am trying to check other PRs (like FreeBSD https://reviews.llvm.org/D128084 ), and clarify how to handle it

File: libcxx/utils/ci/buildkite-pipeline.yml [hunk 2]
```diff
@@ -274,6 +278,13 @@ steps:
     <<: *android_agent
     <<: *common
 
+- group: ':ohos: OpenHarmony'
```
Comment (mordante) [inline libcxx/utils/ci/buildkite-pipeline.yml:281]:
```suggestion
- group: ' OpenHarmony'
```
This looks better IMO.

**Review by mordante — COMMENTED** [review]
> > In general I'm happy with the changes. Our build system has been changed quite a bit recently. Is your build executed?
> 
> Thank you for the review ) Actually, I've checked the build agent status (on buildkite.com), and it is connected, but "This agent has not run any jobs". Looks like I did something wrong (it is my first time of buildkite setup, I am sorry). Maybe there are some guides/examples of build agent setup? I checked this link https://libcxx.llvm.org//AddingNewCIJobs.html#addingnewcijobs and presentation from Louis https://www.youtube.com/watch?v=B7gB6van7Bw , but the whole picture is still too vague for me )
> 
> Now I am trying to check other PRs (like FreeBSD https://reviews.llvm.org/D128084 ), and clarify how to handle it

Nevermind it seems the list with builds has a bad UI. Not showing a scrollbar and showing content below it. The build is in the list and it's running here
https://buildkite.com/llvm-project/libcxx-ci/builds/31749

File: libcxx/utils/ci/buildkite-pipeline.yml (hunk 2 above)
Comment (kpdev) [reply libcxx/utils/ci/buildkite-pipeline.yml]: Fixed

**Comment by kpdev** [comment]
In current build https://buildkite.com/llvm-project/libcxx-ci/builds/31757 I see that `OpenHarmony` and `FreeBSD` jobs are in state "Waiting for agent". I checked my build agent status on buildkite.com, and it is "Connected". May I please ask you to advice what to check first? At this moment I am studying https://buildkite.com/docs/agent/v3 , hope will find some ideas there.

**Comment by philnik777** [comment]
It looks to me like the agent isn't connected correctly. I can't see any OHOS agents in the list.

**Comment by kpdev** [comment]
> It looks to me like the agent isn't connected correctly. I can't see any OHOS agents in the list.

@philnik777 Excuse me, which list do you refer? This one https://buildkite.com/llvm-project/libcxx-ci/builds/31757#018bf894-fbd3-4946-acc8-df414178d60b ?

**Comment by philnik777** [comment]
No, I mean https://buildkite.com/organizations/llvm-project/agents. You might not have access to that though.

**Comment by ldionne** [comment]
@kpdev I don't see any agent token mentioning OpenHarmony, so I don't think there's any way your CI runners are connecting to our BuildKite infra. Can you DM me on Discord so we can set that up?

**Comment by kpdev** [comment]
> @kpdev I don't see any agent token mentioning OpenHarmony, so I don't think there's any way your CI runners are connecting to our BuildKite infra. Can you DM me on Discord so we can set that up?

Thank you. I will write to you there today

**Comment by kpdev** [comment]
Converted to draft till ohos builder setup

**Comment by kpdev** [comment]
> Is there still interest in moving forward with this? I do see OHOS agents connected on BuildKite, so it seems like not much is missing to make progress here.

Hi! Yes, I am quite interested with this ) Will try to assemble all lacking parts this week!

**Pull request closed without merging**